core/templates/dev/head/domain/classifier/PythonProgramTokenizer.js

```diff
@@ -0,0 +1,377 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Tokenizer for Python code.
+ *
+ * IMPORTANT NOTE: The tokenizer is build using Python's own tokenizer module.
+ * These functions are simply translated from Python code to JS code and they
+ * both do same task. The unnecessary code from Python's tokenizer module
+ * has been removed before translating it into JS andcCode relevant to
+ * generating tokens has been kept intact. If Python version changes on Oppia-ml
+ * then changes in Python's tokenizer module must be propagated here.
+ *
+ * Python's tokenizer module for Python version 2.7:
+ * https://github.com/python/cpython/blob/2.7/Lib/tokenize.py
+ */
+
+
+oppia.constant('PythonProgramTokenType', {
+  COMMENT: 'COMMENT',
+  NL: 'NL',
+  STRING: 'STRING',
+  INDENT: 'INDENT',
+  DEDENT: 'DEDENT',
+  ENDMARKER: 'ENDMARKER',
+  NUMBER: 'NUMBER',
+  NAME: 'NAME',
+  OP: 'OP',
+  ERRORTOKEN: 'ERRORTOKEN'
+});
+
+oppia.factory('PythonProgramTokenizer', [
+  '$log', 'PythonProgramTokenType', function($log, PythonProgramTokenType) {
+    var group = function() {
+      return '(' + Array.prototype.join.call(arguments, '|') + ')';
+    };
+
+    var maybe = function() {
+      return group(arguments) + '?';
+    };
+
+    var any = function() {
```

Comment:
Using `groupOfRegEx`, `regExMayBePresent`, `repeatedRegEx` for them respectively.